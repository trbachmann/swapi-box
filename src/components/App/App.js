import React, { Component } from 'react';
import '../../styles/main.scss';
import Navigation from '../Navigation/Navigation';
import { FilmScroll } from '../FilmScroll/FilmScroll';
import { fetchSWData } from '../api/apicalls';
import { addHomeWorldInfo, addSpeciesInfo, selectFilm } from '../helpers/helpers';
import { Display } from '../Display/Display';
import PropTypes from 'prop-types';

class App extends Component {
  constructor() {
    super();
    this.state = {
      people: [],
      planets: [], 
      vehicles: [],
      favorites: [],
      filmToShow: '',
      category: '',
      errorStatus: ''
    }
  }

  getPeople = async () => {
    let people = [];

    try {
      const peopleInfo = await fetchSWData(`https://swapi.co/api/people/`);
      people.push(...peopleInfo.results);
      const peopleWithWorldInfo = await addHomeWorldInfo(people);
      const peopleWithSpeciesAndWorldInfo = await addSpeciesInfo(peopleWithWorldInfo);
      people = peopleWithSpeciesAndWorldInfo;
      this.setState({ people });
    } catch (error) {
      this.setState({errorStatus: error});
    }

  }

  getFilm = async () => {
    try {
      const filmToShow = await selectFilm();
      await this.setState({ filmToShow });
    } catch(error) {
      this.setState({ errorStatus: error });
    }
  }

  componentDidMount = () => {
    this.getFilm();
    this.getPeople();
  }

  updateCategory = (category) => {
    this.setState({category});
  }

  render() {
    let { favorites, category } = this.state;
    return (
      <div className="app">
        <header className="App-header">
        <h1>SWAPI BOX</h1>
        <Navigation updateCategory={this.updateCategory} favorites={favorites.length}/>
        </header>
        <main>
          { this.state.filmToShow !== '' && <FilmScroll film={this.state.filmToShow}/> }
          { this.state.category !== '' && <Display data={this.state[category]} category={category}/>}
        </main>
      </div>
    );
  }
}

Navigation.propTypes = {
  updateCategory: PropTypes.func,
  favorites: PropTypes.number
}

FilmScroll.propTypes = {
  film: PropTypes.object
}

Display.propTypes = {
  data: PropTypes.array,
  category: PropTypes.string
}

export default App;
