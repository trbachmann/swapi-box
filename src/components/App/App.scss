header {
  display: flex;
  justify-content: center;
}
h1 {
  color: #FFE300;
  border: 4px solid #FFE300;
  box-sizing: border-box;
  border-radius: 25px;
  display: inline-block;
  padding: .5em;
  margin-right: auto;
  width: 6.8em;
}
