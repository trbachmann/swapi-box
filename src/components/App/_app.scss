*,
*:after,
*:before {
  box-sizing: border-box;
}

header {
  display: flex;
  justify-content: center;
}

h1 {
  color: $yellow;
  border: 4px solid $yellow;
  box-sizing: border-box;
  border-radius: 25px;
  display: inline-block;
  padding: .5em;
  margin-right: auto;
  margin-left: .8em;
  width: 6.8em;
}

main {
  margin-left: 2%;
  display: grid;
  grid-template-columns: 300px 1fr;
  grid-gap: 3em;
}