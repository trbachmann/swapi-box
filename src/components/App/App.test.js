import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match the snapshot with all data passed in', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should fetch people', () => {

  });

  it('should fetch world information', () => {

  });

  it('should fetch species information', () => {

  });

  it('should update category', () => {

  });
});