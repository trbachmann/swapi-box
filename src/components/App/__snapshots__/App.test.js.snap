// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in 1`] = `
<div
  className="app"
>
  <header
    className="App-header"
  >
    <h1>
      SWAPI BOX
    </h1>
    <Navigation
      favorites={0}
      updateCategory={[Function]}
    />
  </header>
</div>
`;
